models/vanilla_diffusion/dataloader.py

```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+import numpy as np
+
+import torch
+import torchvision.transforms as T
+import torch.nn.functional as F
+import pytorch_lightning as pl 
+from torch.utils.data import Dataset, DataLoader 
+
+class SequenceDataset(Dataset):
```

Comment:
Proposition: We could also separate this script into a `dataloader.py` and a `dataset.py`.

The `dataset.py` would contain the `SequenceDataset` Class and the `dataloader.py` can have the `SequenceDataModule` Class (or a `BaseDataModule` and further subclasses such as `SequenceDataModule` as suggested by @lucapinello)